### PR TITLE
Bar label alignment

### LINKED ISSUE #3160: Bar Label Alignment
### Feature description
Current you cannot change the alignment of the bar labels in the barplot. I noticed this when plotting 90 degree rotated bar labels, they do not align in the center of the bar anymore. 

I would either like to
* somehow automatically improve the bar label alignment for rotated labels
* make the bar label alignment a keyword argument option in the barplot

### For plot types, please add an image of how it should look like

Here's how to get non-centered rotated bar labels. You'll notice the label text is shifted to the left of the center of the bar.

```julia
using CairoMakie, Makie

tbl = (x = [1, 1, 1, 2, 2, 2, 3, 3, 3],
       height = 0.1:0.1:0.9,
       grp = [1, 2, 3, 1, 2, 3, 1, 2, 3],
       grp1 = [1, 2, 2, 1, 1, 2, 1, 1, 2],
       grp2 = [1, 1, 2, 1, 2, 1, 1, 2, 1]
       )

barplot(
        tbl.x, tbl.height,
        dodge = tbl.grp,
        color = tbl.grp,
        bar_labels = :y,
        label_rotation = π/2,
        label_offset = 15,
        axis = (xticks = (1:3, ["left", "middle", "right"]),),
    )
```

RotatedBarLabels

(I also noticed that the ylimit is not ideal by default, the 0.9 label falls outside the axis, but that I can change manually)

I think we can fix this if we can change the `label_aligns` ourselves, which seems to be generated automatically right now.

```
barplot(
        tbl.x, tbl.height,
        dodge = tbl.grp,
        color = tbl.grp,
        bar_labels = :y,
        label_rotation = π/2,
        label_offset = 15,
        axis = (xticks = (1:3, ["left", "middle", "right"]),),
        label_aligns = (:left, :center), # this would probably fix the misalignment
    )
```

### PR DESCRIPTION
**Pull request by matthijscox**
# Description

Fixes #3160

Improved the automatic barplot label alignment and added the attribute `label_align` for custom placement.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist

Done: 
- [x] Added an entry in NEWS.md (for new features and breaking changes)
- [x] Added unit tests for new algorithms, conversion methods, etc.

Not done:
- [x] Added or changed relevant sections in the documentation
- [x] Added reference image tests for new plotting functions, recipes, visual options, etc.

I assume the documentation will automatically show the new attribute.
I considered adding a new reference image, but I think the unit tests cover the behavior well.

So I assume these two items do not need to be checked.

### CODE REVIEW

File: src/basic_recipes/barplot.jl [hunk 1]
```diff
@@ -160,7 +161,35 @@ function stack_grouped_from_to(i_stack, y, grp)
...
+            label_rotation += π/2
+        end
+        s, c = sincos(label_rotation)
+        scale = 1 / max(abs(s), abs(c))
+        align = Vec2f(0.5 - 0.5scale * s, 0.5 - 0.5scale * c)
+        return align
+    elseif is_align_type(label_align)
+        return make_align(label_align)
+    else
+        error("`label_align` needs to be of type NTuple{2, <:Real}, not of type $(typeof(label_align))")
```
Comment (ffreyer) [inline src/basic_recipes/barplot.jl:187]: You can simplify this to something like
```julia
make_align(a::VecTypes{2, <:Real)) = Vec2f(a)
make_align(a::NTuple{2, Symbol}) = to_align(a)
make_align(a) = error(...)
...
else
    return make_align(a)
end
```
(i.e. drop the type check functions)

Comment (matthijscox) [reply src/basic_recipes/barplot.jl:187]: ah good one!

Comment (matthijscox) [reply src/basic_recipes/barplot.jl]: I have pushed a new commit to address your comment

**Comment by matthijscox** [comment]
would someone be willing to run the workflows again, and do a formal review? :)

**Review by SimonDanisch — APPROVED** [review]

**Comment by SimonDanisch** [comment]
Thank you!! :)

**Pull request merged**